File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/paddock/PaddockInformationsForSell.scala





















// Generated on 05/03/2013 10:35:08
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait PaddockInformationsForSell extends TypeDeserializer {
  val typeId = 222

  protected trait Type extends BaseType with Serializable {
    val typeId = PaddockInformationsForSell.typeId

    val guildOwner: String
    val worldX: Short
    val worldY: Short
    val subAreaId: Short
    val nbMount: Byte
    val nbObject: Byte
    val price: Int

    def serialize(buf: Buffer) {
      buf.writeString(guildOwner)
      buf.writeShort(worldX)
      buf.writeShort(worldY)
      buf.writeShort(subAreaId)
      buf.writeByte(nbMount)
      buf.writeByte(nbObject)
      buf.writeInt(price)
    }

  }

  protected def create(guildOwner: String, worldX: Short, worldY: Short, subAreaId: Short, nbMount: Byte, nbObject: Byte, price: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val guildOwner = buf.readString()
    val worldX = buf.readShort()
    val worldY = buf.readShort()
    val subAreaId = buf.readShort()
    val nbMount = buf.readByte()
    val nbObject = buf.readByte()
    val price = buf.readInt()

    create(guildOwner, worldX, worldY, subAreaId, nbMount, nbObject, price)
  }

}

object PaddockInformationsForSell extends PaddockInformationsForSell {
  case class TypeImpl(guildOwner: String, worldX: Short, worldY: Short, subAreaId: Short, nbMount: Byte, nbObject: Byte, price: Int) extends Type
  override def create(guildOwner: String, worldX: Short, worldY: Short, subAreaId: Short, nbMount: Byte, nbObject: Byte, price: Int) = TypeImpl(guildOwner, worldX, worldY, subAreaId, nbMount, nbObject, price)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(guildOwner: String, worldX: Short, worldY: Short, subAreaId: Short, nbMount: Byte, nbObject: Byte, price: Int) = create(guildOwner, worldX, worldY, subAreaId, nbMount, nbObject, price)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/PaddockInformationsForSell.java


// Generated on 05/08/2013 19:38:03
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PaddockInformationsForSell extends NetworkType {
    public static final short TYPE_ID = 222;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public String guildOwner;
    public short worldX;
    public short worldY;
    public short subAreaId;
    public byte nbMount;
    public byte nbObject;
    public int price;
    
    public PaddockInformationsForSell() { }
    
    public PaddockInformationsForSell(String guildOwner, short worldX, short worldY, short subAreaId, byte nbMount, byte nbObject, int price) {
        this.guildOwner = guildOwner;
        this.worldX = worldX;
        this.worldY = worldY;
        this.subAreaId = subAreaId;
        this.nbMount = nbMount;
        this.nbObject = nbObject;
        this.price = price;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeString(guildOwner);
        buf.writeShort(worldX);
        buf.writeShort(worldY);
        buf.writeShort(subAreaId);
        buf.writeByte(nbMount);
        buf.writeByte(nbObject);
        buf.writeInt(price);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        guildOwner = buf.readString();
        worldX = buf.readShort();
        if (worldX < -255 || worldX > 255)
            throw new RuntimeException("Forbidden value on worldX = " + worldX + ", it doesn't respect the following condition : worldX < -255 || worldX > 255");
        worldY = buf.readShort();
        if (worldY < -255 || worldY > 255)
            throw new RuntimeException("Forbidden value on worldY = " + worldY + ", it doesn't respect the following condition : worldY < -255 || worldY > 255");
        subAreaId = buf.readShort();
        if (subAreaId < 0)
            throw new RuntimeException("Forbidden value on subAreaId = " + subAreaId + ", it doesn't respect the following condition : subAreaId < 0");
        nbMount = buf.readByte();
        nbObject = buf.readByte();
        price = buf.readInt();
        if (price < 0)
            throw new RuntimeException("Forbidden value on price = " + price + ", it doesn't respect the following condition : price < 0");
    }
    
}
